src/App.tsx

```diff
@@ -2,28 +1,68 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+type PerPage = '3' | '5' | '10' | '20';
+
+const three: PerPage = '3';
+const five: PerPage = '5';
+const ten: PerPage = '10';
+const twenty: PerPage = '20';
+
+const totalItems: number = 42;
+
+interface Item {
+  id: number;
+  title: string;
+}
+
+const getEndItem = (
+  perPage: number,
+  currentPage: number,
+  amountItems: number,
+): number => {
+  if (perPage * currentPage > amountItems) {
+    return amountItems;
+  }
+
+  return perPage * currentPage;
+};
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState<PerPage>(five);
+  const startItem = +perPage * currentPage - +perPage + 1;
+  const endItem: number = getEndItem(+perPage, currentPage, totalItems);
```

Comment:
The conversion of 'perPage' from a string to a number using the unary plus operator can be unclear. Consider using 'parseInt(perPage, 10)' or 'Number(perPage)' for better readability and to avoid potential issues with non-numeric strings.